### PR TITLE
Fix 375 and 353 for older compilers

### PR DESCRIPTION
**Pull request by mborland**
Add GCC 4.6 to see if other issues with ancient compilers exist

### CODE REVIEW

File: include/boost/test/detail/config.hpp [hunk 1]
```diff
@@ -157,6 +157,17 @@ class type_info;
 # define BOOST_TEST_UNREACHABLE_RETURN(x) BOOST_UNREACHABLE_RETURN(x)
 #endif
 
+//____________________________________________________________________________//
+
+// MSVC <= 12 and GCC <= 4.6 do not allow for defaulted destructors
+// See: https://github.com/boostorg/test/issues/385
+
+#if (defined(BOOST_MSVC) && BOOST_MSVC >= 1900) || (defined(__GNUC__) && !defined(__clang__) && !defined(BOOST_INTEL) && (__GNUC__ >= 5 || __GNUC_MINOR__ >= 7))
```
Comment (Lastique) [inline include/boost/test/detail/config.hpp:165]: You can use `BOOST_GCC` here. It is only defined for genuine gcc.

**Comment by mborland** [comment]
@Lastique I am going to merge this now that the CI cycled. Let me know if you have any remaining issues after the fixes.

**Comment by Lastique** [comment]
> @Lastique I am going to merge this now that the CI cycled. Let me know if you have any remaining issues after the fixes.

Ok, thanks. I'll rerun Boost.Log CI. It'll take time for it to complete.

**Comment by Lastique** [comment]
The CI has passed without errors. Thanks again.

**Pull request merged**